compiler/modules/front.syntax/tests/posTree.bal

```diff
@@ -0,0 +1,585 @@
+        childNodes.push({ name: prefix, pos: () });
+    }
+    childNodes.push(SEMICOLON_NODE);
+    return { childNodes , astNode: decl };
+}
+
+function moduleLevelDefnToNode(ModuleLevelDefn defn) returns SyntaxNode {
+    if defn is ConstDefn {
+        return constDefnToNode(defn);
+    }
+    else if defn is TypeDefn {
+        return typeDefnToNode(defn);
+    }
+    else {
+        return functionDefnToNode(defn);
+    }
+}
+
+function constDefnToNode(ConstDefn defn) returns SyntaxNode {
+    SyntaxNode[] childNodes = moduleLevelDefnPrefix(defn);
+    childNodes.push({ token: "const" });
+    SubsetBuiltinTypeDesc? td = defn.td;
+    if td != () {
+        childNodes.push({ token: td.builtinTypeName });
+    }
+    childNodes.push({ name: defn.name, pos: defn.namePos });
+    childNodes.push({ token: "=" });
+    childNodes.push(exprToNode(defn.expr));
+    childNodes.push(SEMICOLON_NODE);
+    return { childNodes, astNode: defn };
+}
+
+function typeDefnToNode(TypeDefn defn) returns SyntaxNode {
+    SyntaxNode[] childNodes = moduleLevelDefnPrefix(defn);
+    childNodes.push({ token: "type" });
+    childNodes.push({ name: defn.name, pos: defn.namePos });
+    childNodes.push(typeDescToNode(defn.td));
+    childNodes.push(SEMICOLON_NODE);
+    return { childNodes, astNode: defn };
+}
+
+function functionDefnToNode(FunctionDefn defn) returns SyntaxNode {
+    SyntaxNode[] childNodes = moduleLevelDefnPrefix(defn);
+    childNodes.push({ token: "function" });
+    childNodes.push({ name: defn.name, pos: defn.namePos });
+    childNodes.push(functionTypeDescToNode(defn.typeDesc));
+    childNodes.push(...stmtBlockToNodes(defn.body));
+    return { childNodes, astNode: defn };
+}
+
+function moduleLevelDefnPrefix(ModuleLevelDefn defn) returns SyntaxNode[] {
+    SyntaxNode[] childNodes = [];
+    if defn.vis == "public" {
+        childNodes.push({ token: "public", pos: defn.startPos });
+    }
+    return childNodes;
+}
+
+function functionTypeDescToNode(FunctionTypeDesc td) returns SyntaxNode {
+    SyntaxNode[] childNodes = [{ token: "(", pos: td.startPos }];
+    foreach int i in 0 ..< td.params.length() {
+        if i > 0 {
+            childNodes.push({ token: "," });
+        }
+        childNodes.push(functionTypeParamToNode(td.params[i]));
+    }
+    childNodes.push({ token: ")" });
+    TypeDesc retTd = td.ret;
+    // pr-todo: this will mess "returns null" functions
+    if !(retTd is BuiltinTypeDesc && retTd.builtinTypeName == "null") {
+        childNodes.push({ token: "returns" }, typeDescToNode(retTd));
+    }
+    return { childNodes, astNode: td };
+}
+
+function functionTypeParamToNode(FunctionTypeParam param) returns SyntaxNode {
+    SyntaxNode[] childNodes = [typeDescToNode(param.td)];
+    string? name = param.name;
+    if name != () {
+        childNodes.push({ name, pos:param.namePos });
+    }
+    return { childNodes, astNode: param };
+}
+
+function stmtBlockToNodes(StmtBlock block) returns SyntaxNode[] {
```

Comment:
I think the block should be its own NonTerminal (it is in the spec)

Reply:
fixed